### ISSUE TITLE
Delta lake connector, DSR - partition column value is displayed null when queried

### ISSUE THREAD

**Issue by gshenoy123** [prestodb/presto#18570]
Hi,

We are encountering an issue where in delta table partition field value is returned as **NULL** by Presto query.
This results in presto queries which are based on the partitioned fields, returning inconsistent results.
We executed the query using Spark SQL and I can see the value of partition field "ODS" displayed properly.

- Meta store - hive
- Data store - S3
- Data format - Parquet
- Date format - yyyy-MM-dd

> Please note that any field which is chosen as partition is displaying NULL value.

Below are the ingestion and querying code, query snippets, would highly appreciate any pointers.

### Spark

1. Ingestion

```
schema = StructType([
    StructField("SYS_ID", IntegerType(),True), 
	...
    StructField("TRX_DATE", StringType(), True),  # DateType, format - yyyy-MM-dd
    StructField("COUNTRY_CODE", IntegerType(), True)
])

df = spark.read.csv("s3a://data-poc/data/sample/transactions/*", sep=',', header=True, schema=schema)
out_dir_s3 = f"s3a://data-poc/data/sample/transactions/table/"
df.write.format("delta").partitionBy("TRX_DATE").mode("overwrite").save(out_dir_s3)
```

2. Spark SQL

```
%%sparksql
create or replace table trx_poc_1(
    SYS_ID integer,
	...
    TRX_DATE date,
    COUNTRY_CODE integer
)
using delta
partitioned by (TRX_DATE)
location 's3a://data-poc/data/sample/transactions/table/'

%%sparksql
select SYS_ID, ODS from trx_poc_1 limit 1

```

```
SYS_ID		TRX_DATE
90032115	2020-02-10
```

### Presto

I created table using DSR - https://prestodb.io/docs/current/connector/deltalake.html

```
presto> create table hive.default.trx_poc_1(dummy integer) with (external_location='s3a://data-poc/data/sample/transactions/table/');

presto> select SYS_ID, TRX_DATE from trx_poc_1 limit 1
```

```
SYS_ID		TRX_DATE 
90032115	NULL
```

**Comment by agrawalreetika** [comment]
@gshenoy123 Just wanted to confirm, if you have your partition directory structure in this form on s3? 
/delta-path/{partition-columnname}={partition-value} (Ex - s3a://data-poc/data/sample/transactions/table/trx_data={val})

If not, could you please update it and try? And column names are stored in lowercase in metastore so make sure the partition name is in lowercase in s3 too.

**Comment by gshenoy123** [comment]
@agrawalreetika thanks for the reply. Yes the format is as what you mentioned. The s3 table path and partitions listing are depicted as follows - 

```
aws s3 ls s3://data-poc/data/sample/transactions/table
```

**Output** 

> pre TRX_DATA=2022-10-01/
> pre TRX_DATA=2022-10-02/
> pre TRX_DATA=2022-10-03/
> ....

Regarding table creation in hive metastore, we created it using dummy column as per connector document (see reference below) and expected delta connector (DSR) to understand the table structure.  We did the same and can see **desc trx_poc_1** does list down all table columns  and along with partition column - TRX_DATA.

> We created table in Presto using DSR > https://prestodb.io/docs/current/connector/deltalake.html

> As mentioned in the documentation above, we have two catalogue in hive meta store -

> hive - we create schema using catalog and schema prefix hive.default.trx_poc_1
> delta - we query using catalog and schema prefix delta.default.trx_poc_1

Please let know if you require any other info.

**Comment by agrawalreetika** [comment]
@gshenoy123 You are right, metadata would be read via delta transaction logs while using presto-delta connector for the query. But I think you might be getting all the column names in lowercase while querying from Presto. So I think for data partition column name mapping is not happening. 
I have a delta partition table, which I am able to query from the presto-delta connector fine. 
If this doesn't help you can ping me on slack, happy to help to debug this further.
